### ISSUE TITLE
OKX float division by zero

### ISSUE THREAD

**Issue by byte-flylf** [freqtrade/freqtrade#7448]
## Describe your environment

  * Operating system: _centos7___
  * Python Version: __3.10.5___ (`python -V`)
  * CCXT version: _ccxt==1.92.52____ (`pip freeze | grep ccxt`)
  * Freqtrade Version: __freqtrade 2022.8__ (`freqtrade -V` or `docker-compose run --rm freqtrade -V` for Freqtrade running in docker)
  
Note: All issues other than enhancement requests will be closed without further comment if the above template is deleted or not filled out.

## Describe the problem:

*Explain the problem you have encountered*

### Steps to reproduce:

  1. _____
  2. _____
  3. _____
  
### Observed Results:

  * What happened?
  * What did you expect to happen?

### Relevant code exceptions or logs
```
022-09-21 05:59:55,856 - {/root/freqtrade/freqtrade/exchange/exchange.py:1807} - DEBUG - one_call: 2880000000 msecs (a month)
2022-09-21 05:59:55,856 - {/root/freqtrade/freqtrade/exchange/exchange.py:1956} - DEBUG - Fetching pair ETH/USDT:USDT, interval 4h, since 1663704000000 (2022-09-20T20:00:00+00:00) ...
2022-09-21 05:59:55,857 - {/root/freqtrade/freqtrade/exchange/exchange.py:1956} - DEBUG - Fetching pair ETH/USDT:USDT, interval 8h, since 1663704000000 (2022-09-20T20:00:00+00:00) ...
2022-09-21 05:59:55,925 - {/root/freqtrade/freqtrade/exchange/exchange.py:1987} - DEBUG - Done fetching pair ETH/USDT:USDT, interval 4h ...
2022-09-21 05:59:55,966 - {/root/freqtrade/freqtrade/exchange/exchange.py:1987} - DEBUG - Done fetching pair ETH/USDT:USDT, interval 8h ...
2022-09-21 05:59:55,966 - {/root/freqtrade/freqtrade/data/converter.py:33} - DEBUG - Converting candle (OHLCV) data to dataframe for pair ETH/USDT:USDT.
2022-09-21 05:59:55,986 - {/root/freqtrade/freqtrade/data/converter.py:33} - DEBUG - Converting candle (OHLCV) data to dataframe for pair ETH/USDT:USDT.
2022-09-21 05:59:56,009 - {/root/freqtrade/freqtrade/wallets.py:174} - INFO - Wallets synced.
2022-09-21 05:59:56,018 - {/root/freqtrade/.env/lib/python3.10/site-packages/urllib3/connectionpool.py:456} - DEBUG - https://www.okx.com:443 "GET /api/v5/market/books?instId=ETH-USDT-SWAP&sz=1 HTTP/1.1" 200 None
2022-09-21 05:59:56,019 - {/root/freqtrade/freqtrade/exchange/exchange.py:922} - DEBUG - ETH/USDT:USDT checking dry buy-order: price=1321.7, limit=1321.61
2022-09-21 05:59:56,020 - {/root/freqtrade/freqtrade/exchange/exchange.py:1532} - DEBUG - Using cached exit rate for ETH/USDT:USDT.
2022-09-21 05:59:56,020 - {/root/freqtrade/freqtrade/commands/trade_commands.py:21} - ERROR - float division by zero
2022-09-21 05:59:56,020 - {/root/freqtrade/freqtrade/commands/trade_commands.py:22} - ERROR - Fatal exception!
Traceback (most recent call last):
  File "/root/freqtrade/freqtrade/commands/trade_commands.py", line 19, in start_trading
    worker.run()
  File "/root/freqtrade/freqtrade/worker.py", line 74, in run
    state = self._worker(old_state=state)
  File "/root/freqtrade/freqtrade/worker.py", line 114, in _worker
    self._throttle(func=self._process_running, throttle_secs=self._throttle_secs)
  File "/root/freqtrade/freqtrade/worker.py", line 139, in _throttle
    result = func(*args, **kwargs)
  File "/root/freqtrade/freqtrade/worker.py", line 152, in _process_running
    self.freqtrade.process()
  File "/root/freqtrade/freqtrade/freqtradebot.py", line 211, in process
    self.process_open_trade_positions()
  File "/root/freqtrade/freqtrade/freqtradebot.py", line 532, in process_open_trade_positions
    self.check_and_call_adjust_trade_position(trade)
  File "/root/freqtrade/freqtrade/freqtradebot.py", line 596, in check_and_call_adjust_trade_position
    self.execute_trade_exit(trade, current_exit_rate, exit_check=ExitCheckTuple(
  File "/root/freqtrade/freqtrade/freqtradebot.py", line 1561, in execute_trade_exit
    self._notify_exit(trade, order_type, sub_trade=bool(sub_trade_amt), order=order_obj)
  File "/root/freqtrade/freqtrade/freqtradebot.py", line 1584, in _notify_exit
    profit_ratio = trade.calc_profit_ratio(profit_rate, amount, trade.open_rate)
  File "/root/freqtrade/freqtrade/persistence/trade_model.py", line 832, in calc_profit_ratio
    profit_ratio = (1 - (close_trade_value / open_trade_value)) * leverage
ZeroDivisionError: float division by zero
2022-09-21 05:59:56,040 - {/root/freqtrade/freqtrade/commands/trade_commands.py:27} - INFO - worker found ... calling exit
2022-09-21 05:59:56,040 - {/root/freqtrade/freqtrade/rpc/rpc_manager.py:75} - INFO - Sending rpc message: {'type': status, 'status': 'process died'}
2022-09-21 05:59:56,041 - {/root/freqtrade/freqtrade/rpc/rpc_manager.py:81} - DEBUG - Forwarding message to rpc.telegram
2022-09-21 05:59:56,405 - {/root/freqtrade/freqtrade/rpc/rpc_manager.py:81} - DEBUG - Forwarding message to rpc.apiserver
2022-09-21 05:59:56,405 - {/root/freqtrade/freqtrade/freqtradebot.py:144} - INFO - Cleaning up modules ...
```

Note: Please copy/paste text of the messages, no screenshots of logs please.

  ```
  // paste your log here

(.env) [root@testhost freqtrade]# freqtrade show-trades --db-url sqlite:///tradesv3.dryrun.sqlite --trade-ids 36 --print-json
2022-09-21 08:37:56,513 - {/root/freqtrade/freqtrade/configuration/load_config.py:92} - INFO - Using config: user_data/config.json ...
2022-09-21 08:37:56,515 - {/root/freqtrade/freqtrade/loggers.py:132} - INFO - Verbosity set to 0
2022-09-21 08:37:56,515 - {/root/freqtrade/freqtrade/configuration/configuration.py:152} - INFO - Parameter --db-url detected ...
2022-09-21 08:37:56,515 - {/root/freqtrade/freqtrade/configuration/configuration.py:185} - INFO - Using user-data directory: /root/freqtrade/user_data ...
2022-09-21 08:37:56,516 - {/root/freqtrade/freqtrade/configuration/configuration.py:188} - INFO - Using data directory: /root/freqtrade/user_data/data/okx ...
2022-09-21 08:37:56,516 - {/root/freqtrade/freqtrade/configuration/configuration.py:501} - INFO - Parameter --print-json detected ...
2022-09-21 08:37:56,516 - {/root/freqtrade/freqtrade/configuration/configuration.py:501} - INFO - Filtering on trade_ids: ['36']
2022-09-21 08:37:56,516 - {/root/freqtrade/freqtrade/configuration/check_exchange.py:28} - INFO - Checking exchange...
2022-09-21 08:37:56,526 - {/root/freqtrade/freqtrade/configuration/check_exchange.py:56} - INFO - Exchange "okx" is officially supported by the Freqtrade development team.
2022-09-21 08:37:56,526 - {/root/freqtrade/freqtrade/configuration/configuration.py:531} - INFO - Using pairlist from configuration.
2022-09-21 08:37:56,527 - {/root/freqtrade/freqtrade/configuration/config_validation.py:90} - INFO - Validating configuration ...
2022-09-21 08:37:56,532 - {/root/freqtrade/freqtrade/commands/list_commands.py:214} - INFO - Using DB: "sqlite:///tradesv3.dryrun.sqlite"
2022-09-21 08:37:56,587 - {/root/freqtrade/freqtrade/commands/list_commands.py:222} - INFO - Printing 1 Trades: 
[
    {
        "trade_id": 36,
        "pair": "ETH/USDT:USDT",
        "base_currency": "ETH",
        "quote_currency": "USDT",
        "is_open": true,
        "exchange": "okx",
        "amount": 0.8,
        "amount_requested": 0.89987876,
        "stake_amount": 213.7296,
        "strategy": "FFiveMinStrategy",
        "buy_tag": "down >= 0.8",
        "enter_tag": "down >= 0.8",
        "timeframe": 5,
        "fee_open": 0.0002,
        "fee_open_cost": 0.2137296,
        "fee_open_currency": "USDT",
        "fee_close": 0.0002,
        "fee_close_cost": null,
        "fee_close_currency": null,
        "open_date": "2022-09-20 21:45:05",
        "open_timestamp": 1663710305966,
        "open_rate": 1335.81,
        "open_rate_requested": 1335.81,
        "open_trade_value": 1068.4342704,
        "close_date": null,
        "close_timestamp": null,
        "realized_profit": 0.0,
        "close_rate": null,
        "close_rate_requested": 1321.61,
        "close_profit": null,
        "close_profit_pct": null,
        "close_profit_abs": null,
        "trade_duration_s": null,
        "trade_duration": null,
        "profit_ratio": null,
        "profit_pct": null,
        "profit_abs": null,
        "sell_reason": "partial_exit",
        "exit_reason": "partial_exit",
        "exit_order_status": "",
        "stop_loss_abs": 1373.35,
        "stop_loss_ratio": -0.198,
        "stop_loss_pct": -19.8,
        "stoploss_order_id": null,
        "stoploss_last_update": "2022-09-20 21:59:55",
        "stoploss_last_update_timestamp": 1663711195387,
        "initial_stop_loss_abs": 1388.71,
        "initial_stop_loss_ratio": -0.198,
        "initial_stop_loss_pct": -19.8,
        "min_rate": 1321.03,
        "max_rate": 1340.59,
        "leverage": 5.0,
        "interest_rate": 0.0,
        "liquidation_price": 1582.7918937282232,
        "is_short": true,
        "trading_mode": "futures",
        "funding_fees": 0.0,
        "open_order_id": "dry_run_buy_1663711196.009675",
        "orders": [
            {
                "amount": 0.8,
                "safe_price": 1335.81,
                "ft_order_side": "sell",
                "order_filled_timestamp": 1663710374869,
                "ft_is_entry": true,
                "pair": "ETH/USDT:USDT",
                "order_id": "dry_run_sell_1663710305.887764",
                "status": "closed",
                "average": 1335.81,
                "cost": 1068.648,
                "filled": 0.8,
                "is_open": false,
                "order_date": "2022-09-20 21:45:05",
                "order_timestamp": 1663710305000,
                "order_filled_date": "2022-09-20 21:46:14",
                "order_type": "limit",
                "price": 1335.81,
                "remaining": 0.0
            },
            {
                "amount": 0.0,
                "safe_price": 1321.61,
                "ft_order_side": "buy",
                "order_filled_timestamp": null,
                "ft_is_entry": false,
                "pair": "ETH/USDT:USDT",
                "order_id": "dry_run_buy_1663711196.009675",
                "status": "open",
                "average": 1321.61,
                "cost": 0,
                "filled": 0.0,
                "is_open": true,
                "order_date": "2022-09-20 21:59:56",
                "order_timestamp": 1663711196000,
                "order_filled_date": null,
                "order_type": "limit",
                "price": 1321.61,
                "remaining": 0.0
            }
        ]
    }
]

  ```

**Comment by xmatthias** [comment]
The log is unfortunately truncated a little too much at the top - so it's unclear if this was a partial exit.

Assuming it was - i think this was fixed in 4d69df08dde96d067d9daebbf2db3a61a24dd44a.

**Comment by byte-flylf** [comment]
fixed in https://github.com/freqtrade/freqtrade/commit/4d69df08dde96d067d9daebbf2db3a61a24dd44a.
